### DISCUSSION TITLE
Same filename issue

### DISCUSSION THREAD

**Discussion by mosolab** [obss/sahi#721] [Q&A]
Hi guys! 

Thank you for this great tool! 

Could you support me? I have and issue during sliced coco dataset preparation.

Original dataset has such structure:
OBJ_1/IMG_01.JPG
OBJ_1/IMG_02.JPG
...
OBJ_2/IMG_01.JPG
OBJ_2/IMG_02.JPG
...

And there is one single coco.json for this folder and file structure.

I try to slice it to some output folder.
And then got sliced images only for last OBJ_N in original coco.json.

Would it be possible to somehow handle such folder and file structure.

Thank you in advance!

**Comment by fcakyon** [comment]
Thanks for the support!

Can you paste your full code so that we can understand you error better?

Also have you updated your sahi to latest version?

Can you also paste the error message you are having?

**Reply by mosolab** [reply]
Thanks for reply!
And sorry for long reply. I updated to latest version.
I checked again and see right now that many errors and warnings regarding wrong annotation and files are skipped:
 
```
11/08/2022 16:45:04 - ERROR - shapely.geos -   TopologyException: Input geom 0 is invalid: Self-intersection at 1132.0488377433214 144.26051045129904
11/08/2022 16:45:04 - INFO - shapely.geos -   Self-intersection at or near point 1132.0488377433214 144.26051045129904
11/08/2022 16:45:04 - WARNING - sahi.slicing -   Invalid annotation found, skipping this image:
```

So now question little bit different.
Would it be possible to check with sahi  what is wrong with annotation? 
Original dataset labels contain both boxx and segmentation.

**Reply by fcakyon** [reply]
it seems there are some invalid segmentation polygons in the dataset, maybe you can visualize the annotations using fifty-one. you can try `sahi coco fiftyone` command: https://github.com/obss/sahi/blob/main/docs/fiftyone.md
